src/prototype_creep_resources.js

```diff
@@ -486,7 +486,7 @@ Creep.prototype.transferMy = function() {
 };
 
 Creep.prototype.getEnergy = function() {
-  if (this.carry.energy == this.carryCapacity) {
+  if (this.carry.energy > 0) {
```

Comment:
I tried something out with a state machine in `role_mineral`, ended in a mess :-)

Yes, sure, a state machine could make it more explicit.